### ISSUE TITLE
Replace with SCTP

### ISSUE THREAD

**Issue by djugei** [amethyst/laminar#247]
This might seem like a troll, but [SCTP](https://en.wikipedia.org/wiki/Stream_Control_Transmission_Protocol) is a rather mature protocol that is widely used in WebRTC and has all the features listed on the README. There is even some (incomplete) [rust implementations](https://crates.io/search?q=sctp).

Specifically SCTP has heartbeats, any combination of reliability and ordering, transmission timeouts and multiple streams just like laminar. It also has stuff like multipathing/multihoming (especially relevant for mobile!), DDOS-resistant handshakes, mature reference implementations and years of engineering.

Networking protocols are really hard to get right. It is kinda easy to get the packets flowing, but the details are a lot of work. Especially congestion control is crazy complicated, in fact complicated enough that even the linux-kernel sometimes has bugs in it.

Building a mature SCTP implementation in rust seems like it would remove a nontrivial amount of duplication of work, plus the benefit of having free WebRTC-compatibility and other synergies.

If this is not the way this project wants to go then please at least add a section to the README that contrasts laminar to other network protocols to counter nih-claims.

clarification: sctp would be deployed as [SCTP over UDP](https://tools.ietf.org/html/rfc6951), thereby avoiding problems with middleboxes.

**Comment by erlend-sh** [comment]
Interesting. We welcome input from the SCTP implementers as well, @phsym @simmons @danclive

**Comment by TimonPost** [comment]
We've covered this slot of times before, discord history is full of it and Reddit as well. Please have a look at the following [discussion](https://www.reddit.com/r/rust/comments/bctdtj/laminar_020_reliable_udp_game_protocol_amethyst/?utm_medium=android_app&utm_source=share).

**Comment by djugei** [comment]
@TimonPost Then lets use this issue as a basis for a document that contrasts SCTP and laminar so people can more easily evaluate which usecases fit what protocol.

Those discussions do not seem very... full of content though. Its basically just a mention of sctp, a mention of how layer-3-sctp probably will not work and a response that sctp is usually deployed on top of udp. no further followup happening.

**Comment by TimonPost** [comment]
I totally understand your concerns. The discussions are all over the place (Github, Reddit, discord), I would have like to get all the differences listed, but we decided that we would not spend any time on it anymore. Because it is always the same thing, we mention laminar, people are coming up with millions of other alternatives, sure QUIC is an other example, probably one I would go for over SCTP.
Fortunately, we fixed that problem with our new [networking implementation](https://github.com/amethyst/amethyst/pull/1917), where everyone is able to implement whatever protocol they want for their game. Laminar is not about whether SCTP or QUICK is better, it is about building an protocol according to what gaffer wrote in a series of block posts. So, in the end, there is totally no need for discussion. And we will probably have a default QUIC implementation in amethyst as a separate transport layer provider.
